Detectors/ZDC/workflow/src/reco-reader-workflow.cxx

```diff
@@ -0,0 +1,47 @@
+// Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+// See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// \file digits-reader-workflow.cxx
+/// \brief Implementation of ZDC digits reader
+///
+/// \author ruben.shahoyan@cern.ch
+
+#include "Framework/CallbackService.h"
+#include "Framework/ControlService.h"
+#include "Framework/ConfigParamRegistry.h"
+#include "Framework/Task.h"
+#include "ZDCWorkflow/RecoReaderSpec.h"
+#include "CommonUtils/ConfigurableParam.h"
+
+using namespace o2::framework;
+
+// we need to add workflow options before including Framework/runDataProcessing
+void customize(std::vector<ConfigParamSpec>& workflowOptions)
+{
+  // option allowing to set parameters
+
+  std::vector<o2::framework::ConfigParamSpec> options{};
+
+  std::string keyvaluehelp("Semicolon separated key=value strings");
+  options.push_back(ConfigParamSpec{"configKeyValues", VariantType::String, "", {keyvaluehelp}});
+  std::swap(workflowOptions, options);
+}
+
+#include "Framework/runDataProcessing.h"
+
+WorkflowSpec defineDataProcessing(const ConfigContext& ctx)
+{
+  WorkflowSpec specs;
+
+  DataProcessorSpec producer = o2::zdc::getRecoReaderSpec();
+  specs.push_back(producer);
+  return specs;
+}
```

Comment:
You need these lines in all workflows with readers which may read MC-produced inputs:

https://github.com/AliceO2Group/AliceO2/blob/524a937eacf610b98e6ba851fe36f826982ccd08/Detectors/GlobalTrackingWorkflow/readers/src/irframe-reader-workflow.cxx#L45
https://github.com/AliceO2Group/AliceO2/blob/524a937eacf610b98e6ba851fe36f826982ccd08/Detectors/GlobalTrackingWorkflow/readers/src/irframe-reader-workflow.cxx#L29

https://github.com/AliceO2Group/AliceO2/blob/524a937eacf610b98e6ba851fe36f826982ccd08/Detectors/GlobalTrackingWorkflow/src/primary-vertexing-workflow.cxx#L99

Reply:
It works! Thanks